File: Data Structures & Algorithms/non-overlapping-intervals/submission-5.java
class Solution {
    public int eraseOverlapIntervals(int[][] intervals) {
        Arrays.sort(intervals, (a,b)->Integer.compare(a[0],b[0]));
        int minRemovals=0;
        int prevEnd=intervals[0][1];
        for(int i=1; i<intervals.length; i++){
            int start=intervals[i][0];
            int end=intervals[i][1];
            if(start<prevEnd){
                minRemovals+=1;
                prevEnd=Math.min(prevEnd, end);
            }
            else{
                prevEnd=intervals[i][1];
            }
        }
        return minRemovals;
    }
}

File: Data Structures & Algorithms/longest-common-prefix/submission-5.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        commonPrefix=strs[0]
        for string in strs[1:]:
            i=0
            while i<min(len(string), len(commonPrefix)) and string[i]==commonPrefix[i]:
                i+=1
            commonPrefix=string[:i]
        return commonPrefix
        